### ISSUE TITLE
Vite: Enable history mode in Dev Server

### ISSUE THREAD

**Issue by dschmidt** [owncloud/web#8098]
### Expected behaviour
The oC Web instance served by Vite at `https://host.docker.internal:9201/` or `http://host.docker.internal:8080` should run with Vue Router in `history` mode.

### TODOs

- [x] Add `<base href="/">` to `<head>` section of `index.html` (easy) and to `oidc-callback(-silent).html` (unknown).
- [x] Make history mode in Vite work (c.f. https://github.com/vitejs/vite/issues/11282)

### Ideas
Adding the base tag could be done like this for `index.html`, but it doesn't work for `oidc-callback(-silent).html`:
```typescript
        {
          name: 'add-base',
          transformIndexHtml: {
            transform(html, { bundle }) {
              if (bundle) {
                return
              }
              return [
                {
                  tag: 'base',
                  attrs: {
                    'href': '/'
                  }
               }
              ]
            }
          }
        }
```

**Comment by dschmidt** [comment]
First task is solved for CERN in #9450.

Second one can probably be solved like this: https://github.com/vitejs/vite/issues/2415#issuecomment-1526393135

**Comment by dschmidt** [comment]
https://github.com/vitejs/vite/pull/14142 🥳

**Comment by dschmidt** [comment]
Fixed in #9720
